lib/collection/src/operations/verification/update.rs

```diff
@@ -317,6 +327,131 @@ async fn check_multivectors_limits_update(
+        })
+        .collect();
+
+    (!sparse_max_size.is_empty()).then_some(sparse_max_size)
+}
+
+async fn check_sparse_vector_limits_update(
+    point_insert: &UpdateVectors,
+    sparse_config: &StrictModeSparseConfig,
+) -> Result<(), CollectionError> {
+    let Some(sparse_max_size_by_name) = sparse_limits(sparse_config).await else {
+        return Ok(());
+    };
+
+    for point in &point_insert.points {
+        check_sparse_vecstruct_limit(&point.vector, &sparse_max_size_by_name)?;
+    }
+
+    Ok(())
+}
+
+async fn check_sparse_vector_limits_insert(
+    point_insert: &PointInsertOperations,
+    sparse_config: &StrictModeSparseConfig,
+) -> Result<(), CollectionError> {
+    let Some(sparse_max_size_by_name) = sparse_limits(sparse_config).await else {
+        return Ok(());
+    };
+
+    match point_insert {
+        PointInsertOperations::PointsBatch(batch) => match &batch.batch.vectors {
+            BatchVectorStruct::Named(named_batch_vectors) => {
+                for (name, vectors) in named_batch_vectors {
+                    for vector in vectors {
+                        check_named_sparse_vec_limit(name, vector, &sparse_max_size_by_name)?;
+                    }
+                }
+            }
+
+            BatchVectorStruct::Single(_)
+            | BatchVectorStruct::Document(_)
+            | BatchVectorStruct::MultiDense(_)
+            | BatchVectorStruct::Image(_)
+            | BatchVectorStruct::Object(_) => {}
+        },
+        PointInsertOperations::PointsList(list) => {
+            for point_struct in &list.points {
+                match &point_struct.vector {
+                    VectorStruct::Named(named_vectors) => {
+                        for (name, vector) in named_vectors {
+                            check_named_sparse_vec_limit(name, vector, &sparse_max_size_by_name)?;
+                        }
+                    }
+                    VectorStruct::Single(_) => {}
+                    VectorStruct::MultiDense(_) => {}
+                    VectorStruct::Document(_) => {}
+                    VectorStruct::Image(_) => {}
+                    VectorStruct::Object(_) => {}
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn check_sparse_vecstruct_limit(
+    vector: &VectorStruct,
+    sparse_max_size_by_name: &TinyMap<String, usize>,
+) -> Result<(), CollectionError> {
+    match vector {
+        VectorStruct::Named(named) => {
+            for (name, vec) in named {
+                check_named_sparse_vec_limit(name, vec, sparse_max_size_by_name)?;
+            }
+            Ok(())
+        }
+        VectorStruct::Single(_) => Ok(()),
+        VectorStruct::MultiDense(_) => Ok(()),
+        VectorStruct::Document(_) => Ok(()),
+        VectorStruct::Image(_) => Ok(()),
+        VectorStruct::Object(_) => Ok(()),
+    }
+}
+
+fn check_named_sparse_vec_limit(
+    name: &str,
+    vector: &Vector,
+    sparse_max_size_by_name: &TinyMap<String, usize>,
```

Comment:
Can we replace `String` with `&str` here (and in some other places) to further reduce the number of clones we need?